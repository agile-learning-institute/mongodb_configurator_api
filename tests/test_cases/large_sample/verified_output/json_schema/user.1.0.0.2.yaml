description: A user collection for testing the schema system
type: object
properties:
  _id:
    description: The unique identifier for a user
    type: string
    pattern: ^[0-9a-fA-F]{24}$
  user_name:
    description: Username
    type: string
    pattern: ^\S{1,40}$
  first_name:
    description: Users First Name
    type: string
    pattern: ^\S{1,40}$
  last_name:
    description: Users Last Name
    type: string
    pattern: ^\S{1,40}$
  status:
    description: document status
    type: string
    enum:
    - draft
    - active
    - archived
  last_saved:
    description: The last time this document was saved
    type: object
    properties:
      from_ip:
        description: Http Request remote IP address
        type: string
        pattern: ^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$
      by_user:
        description: ID Of User
        type: string
        pattern: ^\S{1,40}$
      at_time:
        description: The date-time when last updated
        type: string
        format: date-time
      correlation_id:
        description: The logging correlation ID of the update transaction
        type: string
        pattern: ^\S{1,40}$
    additionalProperties: false
    required:
    - from_ip
    - by_user
    - at_time
    - correlation_id
required:
- _id
- user_name
- status
- last_saved
additionalProperties: false
