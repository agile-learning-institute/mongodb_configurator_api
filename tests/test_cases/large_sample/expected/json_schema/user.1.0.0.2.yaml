type: object
description: A user collection for testing the schema system
properties:
  _id:
    type: string
    description: The unique identifier for a user
    pattern: "^[0-9a-fA-F]{24}$"
  user_name:
    type: string
    description: Username
    pattern: "^[^\\s]{4,40}$"
  first_name:
    type: string
    description: Users First Name
    pattern: "^[^\\s]{4,40}$"
  last_name:
    type: string
    description: Users Last Name
    pattern: "^[^\\s]{4,40}$"
  status:
    type: string
    description: document status
    enum:
      - draft
      - active
      - archived
  last_saved:
    title: Breadcrumb
    description: The last time this document was saved
    type: object
    properties:
      from_ip:
        description: Http Request remote IP address
        type: string
        pattern: "^[^\\s]{4,40}$"
      by_user:
        description: ID Of User
        type: string
        pattern: "^[^\\s]{4,40}$"
      at_time:
        description: The date-time when last updated
        type: string
        format: date-time
      correlation_id:
        description: The logging correlation ID of the update transaction
        type: string
        pattern: "^[^\\s]{4,40}$" 
    required:
      - from_ip
      - by_user
      - at_time
      - correlation_id
    additionalProperties: false
required:
  - _id
  - user_name
  - status
  - last_saved
additionalProperties: false